assets/js/vue-apps/form-components/address-lookup.vue

```diff
@@ -82,23 +78,21 @@ export default {
             } catch (e) {} // eslint-disable-line no-empty
         }
 
-        if (this.i18n) {
-            try {
-                this.copy = JSON.parse(this.i18n);
-            } catch (e) {} // eslint-disable-line no-empty
-        }
-
         const $form = $(this.$el)
             .parents('form')
             .find('input[type="submit"]');
-        const that = this;
-        $form.on('click', function() {
-            if (
-                that.candidates.length === 0 &&
-                that.postcode &&
-                that.currentState !== states.NotRequired
-            ) {
-                alert(that.copy.fields.address.warningForIncomplete);
+
+        function shouldWarn() {
+            return (
+                this.candidates.length === 0 &&
+                this.postcode &&
+                this.currentState !== states.NotRequired
+            );
+        }
+
+        $form.on('click', () => {
+            if (shouldWarn()) {
+                alert(this.$t('address.warningForIncomplete'));
```

Comment:
Does, er, `this` definitely resolve properly here? I had issues with it originally (though I guess the arrow function might have solved it?).

Reply:
Yeah resolves OK, but I introduced an error in the `shouldWarn` method so double checking this now.